### DISCUSSION TITLE
Perform Aggregation and populate after grouping

### DISCUSSION THREAD

**Discussion by sambitraze** [directus/directus#12638] [Archive]
I have two task
1)First is to perform aggregation like sum or avg on relation fields:
`https://mesh.kodagu.today/items/Courses_Enroll?fields=course.*,*`

I want to do SUM of all prices: data.course.price
ps: it would be better if we have this included in docs we have very less data on nested API calls.

2)Second to group either accord to relation ID or some field inside the related(nested) field in this case let's say I want to group the enrolments by course id and then return the response with populated field 

ex :  { {course: 1, transaction: 1}, {course: 1, transaction: 2}, {course: 1, transaction: 3}}
 this after the groupingnd population should give
{course1 :[{course: 1, transaction: 1}, {course: 1, transaction: 2}], course2:[{course: 1, transaction: 3}]} 
instead of just giving 
{course1 :2, course2:1}

**Comment by rijkvanzanten** [comment]
> I want to do SUM of all prices: data.course.price

Couldn't you use SUM on the Courses collection in that case? 🤔 

`https://mesh.kodagu.today/items/Courses?aggregate[sum]=price`

**Reply by sambitraze** [reply]
no, the course_enrol is the collection for all enrolment done for each course by a user so I want sum the money lets say profits per course

**Reply by sambitraze** [reply]
@rijkvanzanten any other suggestion?

**Reply by rijkvanzanten** [reply]
No, I think a custom endpoint would be the next best step 👍🏻
